implementation/src/main/java/io/smallrye/jwt/auth/principal/KeyLocationResolver.java

```diff
@@ -157,21 +160,36 @@ private PublicKey tryAsJWK(JsonWebSignature jws) throws UnresolvableKeyException
     }
 
     private void loadContents() throws Exception {
-        final String location = authContextInfo.getPublicKeyLocation();
-        if (location.startsWith("https:")) {
-            httpsJwks = new HttpsJwks(location);
+        final URI location = URI.create(authContextInfo.getPublicKeyLocation());
+
+        if ("https".equals(location.getScheme())) {
+            httpsJwks = new HttpsJwks(authContextInfo.getPublicKeyLocation());
             httpsJwks.setDefaultCacheDuration(authContextInfo.getJwksRefreshInterval().longValue() * 60L);
             return;
         }
 
-        StringWriter contents = new StringWriter();
-        final InputStream is;
-        if (location.startsWith("classpath:") || location.indexOf(':') < 0) {
-            is = getAsResource(location);
+        InputStream is = null;
+
+        if (location.getScheme() != null) {
+            if ("classpath".equals(location.getScheme())) {
+                is = getAsClasspathResource(location.getSchemeSpecificPart());
+            } else if ("file".equals(location.getScheme())) {
+                is = getAsFileSystemResource(location.getRawSchemeSpecificPart());
+            } else {
+                is = new URL(authContextInfo.getPublicKeyLocation()).openStream();
+            }
         } else {
-            URL locationURL = new URL(location);
-            is = locationURL.openStream();
+            is = getAsClasspathResource(authContextInfo.getPublicKeyLocation());
+            if (is == null) {
+                is = getAsFileSystemResource(authContextInfo.getPublicKeyLocation());
```

Comment:
As rightfully pointed out in #96, the spec says

> Relative or non-URL paths supplied as the location are resolved in the following order:
>  1. `new File(location)`
>  2. `Thread.currentThread().getContextClassLoader().getResource(location)`

But we first attempt option 2, and only then option 1. Should we swap them?

Reply:
@Ladicek I thought I'd avoid the extra penalty until the TCK enforces it, as it is highly unlikely that the same named resource would exist both on the classpath and the filesystem, having it on the classpath is more likely. I don't mind much swtching the order though for 2.0.2, will open an issue.